honeybee/room.py

```diff
@@ -558,11 +575,16 @@ def solve_adjacency(rooms, tolerance):
                             if not isinstance(face_2.boundary_condition, Surface):
                                 if face_1.geometry.is_centered_adjacent(
                                         face_2.geometry, tolerance):
-                                    face_1.set_adjacency(face_2)
+                                    adj_aps, adj_drs = face_1.set_adjacency(face_2)
+                                    adjacent_faces.append((face_1, face_2))
+                                    adjacent_apertures.extend(adj_aps)
+                                    adjacent_doors.extend(adj_drs)
                                     break
             except IndexError:
                 pass  # we have reached the end of the list of zones
 
+        return adjacent_faces, adjacent_apertures, adjacent_doors
```

Comment:
@chriswmackey this is a generic comment about how to write methods / functions and if it is fine to return multiple values. I don't have the correct answer but I know this will be an issue in the long-term.

From Python point-of-view this is totally fine as you are returning a tuple. From my experience when you move to returning multiple values from a method it doesn't scale well and will make breaking changes in the future.

What will happen, and we had this for the legacy several times, is for an upcoming change we will need to add or remove an output to a method. This will potentially break all the code where this method is called. A possible solution is to enforce returning a Named Tuple of a Dictionary with key / values. That means adding a new output won't break the current use cases. Dictionaries will add an overhead but in the long term it makes the maintenance and development easier.

Related: https://stackoverflow.com/a/61636/4394669

Reply:
I would agree that this is a case where a dictionary is best much like what we do for [comfort functions n ladybug-comfort](https://github.com/ladybug-tools/ladybug-comfort/blob/master/ladybug_comfort/pmv.py#L88-L96). Let's agree that it's worth considering using this dictionary approach any time there are more than 4 outputs being returned or there's even a slight chance that the method may output more variables in the future.